setup/src/setup-haskell.ts

```diff
@@ -33,21 +35,47 @@ export default async function run(
         exec('stack', ['setup', opts.ghc.resolved])
       );
 
-    if (opts.cabal.enable)
-      await core.group('Setting up cabal', async () => {
-        await exec('cabal', ['user-config', 'update'], {silent: true});
-        const configFile = await cabalConfig();
-
-        if (process.platform === 'win32') {
-          fs.appendFileSync(configFile, 'store-dir: C:\\sr\n');
-          core.setOutput('cabal-store', 'C:\\sr');
-        } else {
-          core.setOutput('cabal-store', `${process.env.HOME}/.cabal/store`);
-        }
-
-        await exec('cabal user-config update');
-        if (!opts.stack.enable) await exec('cabal update');
-      });
+    if (opts.cabal.enable) core.info('Loading cache...');
+    await exec('cabal', ['freeze'], {silent: true});
```

Comment:
I'm hesitant to want to run cabal freeze by default. Perhaps we should check for a `cabal.project.freeze` and a `stack.yaml.lock`  by default, but allow users to override this by specifying an option like `cache-keys` that works exactly as `key` + `restore-keys` does on actions/cache now.

Then we can just split on newlines for the `cache-keys` and keep the standard  `os-$ghc-$hash-$sha` logic as the default. This also has the benefit of being.

We will also probably want to support something like `cache-paths` in case some people don't want to cache `dist-newstyle` in addition to `cabal-root` (or don't want to cache `stack-root`, or...)